lib/surgex/date_time/date_time.ex

```diff
@@ -30,13 +35,13 @@ if Code.ensure_loaded?(Timex) do
       end
     end
 
-    defp apply_timezone(datetime, timezone) do
+    defp apply_timezone(datetime, timezone, ambiguous_hour_pref) do
       case Timex.to_datetime(datetime, timezone) do
         %DateTime{} = datetime_in_timezone ->
           {:ok, datetime_in_timezone}
 
         %Timex.AmbiguousDateTime{} = datetime ->
-          {:ok, datetime.after}
+          {:ok, datetime |> Map.get(ambiguous_hour_pref)}
```

Comment:
Since we want to access a field in a struct, perhaps `apply` would be better here?